### PR TITLE
Display IAPs in Account Overview, Billing page and Emails and Marketing consents

### PR DESCRIPTION
**Pull request by andrewHEguardian**
## What does this change?

Adjust the copy and design for IAP product holding on the account overview page and billing page. Get correct Soft Opt Ins for IAPs and refactor to show them on Email and Marketing page.

## How to test

With an IAP product associated with an account test the following:
Account Overview:
Navigate to `/?withFeature=appSubscriptions` and see the App Subscription product card.

Billing:
Go to Billing page and see the App Subscription box.

Email and Marketing:
Go to Email and Marketing and see the following 3 supporter exclusive consents:
- Your subscription/support
- Supporter newsletter
- Similar Guardian products

## Images
image

image

image

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

**Comment by andrewHEguardian** [comment]
[IAP Soft-Opt-In: MMA to display IAPs in Product Card](https://trello.com/c/oqRDD68J/3907-iap-soft-opt-in-mma-to-display-iaps-in-product-card)

File: client/components/mma/accountoverview/InAppPurchaseCard.tsx [hunk 1]
```diff
@@ -16,13 +17,37 @@ const productTitleCss = css`
...
 }) => {
 	return (
 		<Stack space={3}>
-			{props.inAppPurchase.cancellationTimestamp && (
-				<InfoSummary message="ToDo: Your App Subscription Was cancelled" />
+			{inAppPurchase.cancellationTimestamp && (
+				<InfoSummary
+					message={cancelledAppSubscriptionMessage(
+						inAppPurchase.cancellationTimestamp,
+					)}
```
Comment (jamesmockett) [inline client/components/mma/accountoverview/InAppPurchaseCard.tsx:49]: You could use the `context` property for the supporting part of the message which would make things look a little nicer.

https://guardian.github.io/csnx/?path=/story/source-react-components-development-kitchen_info-summary--with-context

File: client/components/mma/billing/Billing.tsx [hunk 2]
```diff
@@ -306,7 +305,31 @@ function renderProductBillingInfo([mmaCategory, productDetails]: [
...
+				`}
+			>
+				App Subscription
+			</h2>
+			<div
+				css={css`
+					${textSans.medium()};
+					border: 1px solid ${neutral[20]};
+					display: flex;
+					flex-wrap: wrap;
```
Comment (jamesmockett) [inline client/components/mma/billing/Billing.tsx:323]: It doesn't look like this needs the flex properties as it only has text content?

Comment (andrewHEguardian) [reply client/components/mma/billing/Billing.tsx:323]: True, was copied from the other billing tables

File: client/components/mma/billing/Billing.tsx [hunk 3]
```diff
@@ -306,7 +305,29 @@ function renderProductBillingInfo([mmaCategory, productDetails]: [
...
+			</h2>
+			<div
+				css={css`
+					${textSans.medium()};
+					border: 1px solid ${neutral[20]};
+					margin: ${space[5]}px 0;
+					padding: ${space[3]}px;
+				`}
+			>
+				To change your payment setup, please contact Apple (for iOS), or
```
Comment (paulbrown1982) [inline client/components/mma/billing/Billing.tsx:326]: Don't we know whether the IAP is an Apple or Google one?

Comment (andrewHEguardian) [reply client/components/mma/billing/Billing.tsx:326]: Not that I can see - unless we can tell from the ProductID?
eg response
     ```
   {
            "subscriptionId": "",
            "from": "2023-01-11T11:05:20.000Z",
            "to": "2023-01-11T12:08:02.000Z",
            "valid": false,
            "gracePeriod": false,
            "autoRenewing": false,
            "productId": "uk.co.guardian.gla.1month.2018May.withFreeTrial"
        }
```

Comment (paulbrown1982) [reply client/components/mma/billing/Billing.tsx:326]: Yes that product ID is the SKU basically, which is unique. Maybe we can have a mapping table / regex in MDAPI to make it useful for display @jonflynng ?

Comment (paulbrown1982) [reply client/components/mma/billing/Billing.tsx:326]: These are the mappings: https://docs.google.com/spreadsheets/d/1I8XZnirRkIQ1RDAq9iSqVC5B9UzFqV4iJnC_dTfydfE/

Comment (andrewHEguardian) [reply client/components/mma/billing/Billing.tsx:326]: Will merge this PR, then discuss with Daniel B if there are other places we could use this info and raise a new PR

**Comment by andrewHEguardian** [comment]
[IAP Soft-Opt-In: Marketing Preferences to show Soft-opt-in/out for IAPs](https://trello.com/c/TwlEFYVv/3909-iap-soft-opt-in-marketing-preferences-to-show-soft-opt-in-out-for-iaps)

**Pull request merged**